tf-psa-crypto/tests/suites/test_suite_psa_crypto.data

```diff
@@ -7626,11 +7633,11 @@ depends_on:MBEDTLS_THREADING_PTHREAD
 concurrently_generate_keys:PSA_KEY_TYPE_RAW_DATA:9:PSA_KEY_USAGE_EXPORT:0:PSA_ERROR_INVALID_ARGUMENT:0:8:5
 
 PSA concurrent key generation: raw data, (MBEDTLS_CTR_DRBG_MAX_REQUEST + 1) * 8 bits
-depends_on:MBEDTLS_THREADING_PTHREAD
+depends_on:MBEDTLS_THREADING_PTHREAD:!MBEDTLS_PSA_STATIC_KEY_SLOTS
```

Comment:
It's been a while since I made this change so I don't remember exactly what it was the problem there (you are totally right: no documentation on this means that even I cannot clearly explain why I did this, sorry). Guessing from the description of the test I would say that the problem here was that these tests are trying to generate a large `PSA_KEY_TYPE_RAW_DATA` and this is unlikely to be successful when static key slot buffers are used.
First of all I will verify that my hypothesis is correct, then I will replace the `depends_on` from `!MBEDTLS_PSA_STATIC_KEY_SLOTS` to something like `MBEDTLS_PSA_STATIC_KEY_SLOT_BUFFER_SIZE>x`

Reply:
Question: I was thinking to improve this by adding the following `depends_on` to the `PSA concurrent key generation: raw data, (MBEDTLS_CTR_DRBG_MAX_REQUEST + 1) * 8 bits` test:

`MBEDTLS_PSA_STATIC_KEY_SLOT_BUFFER_SIZE > MBEDTLS_CTR_DRBG_MAX_REQUEST + 1`.

However `generate_test_code.py` complains that this is an `Invalid dependency`. I suspect that this might be connected to symbols' expansion. Do you have any suggestion on how to implement this?

Reply:
Question: checking `generate_test_code.py` it seems to me that conditional `depends_on` should have the form: 
`<C_symbol> <operator> <numerical value>`. 
I wanted to change the line you commented as follows
```suggestion
depends_on:MBEDTLS_THREADING_PTHREAD:MBEDTLS_PSA_STATIC_KEY_SLOT_BUFFER_SIZE > MBEDTLS_CTR_DRBG_MAX_REQUEST + 1
```
but of course the Python script complains that this is an `Invalid dependency`. Do you have any suggestion on how I can express this dependency here?

Reply:
> checking generate_test_code.py it seems to me that conditional depends_on should have the form:
> `<C_symbol> <operator> <numerical value>`

Indeed. That is annoying.

The only thing that `generate_test_code.py` does with `MBEDTLS_PSA_STATIC_KEY_SLOT_BUFFER_SIZE > MBEDTLS_CTR_DRBG_MAX_REQUEST + 1` is to inject it into an `#if` line. So we could actually relax this condition. For an entry in a `depends_on` clause, all `generate_test_code.py` wants to know is whether this is a bare symbol (turned into `defined(…)`), an exclamation point followed by a bare symbol (turned into `!defined(…)`), or something else (turned into just `(…)`).

We could in principle allow anything and let the C preprocessor complain if the expression is invalid. I think this would be too extreme because the feedback on errors (e.g. typing `FOO;BAR` instead of `FOO:BAR`) is pretty poor: you'd get an error when compiling the `.c` file, with a meaningless error line number in `main_test.function` (there's no easy way to emit a `#line` statement referring to the `.data` file). But we could at least allow a larger subset. For example:

* must contain a relation operator
* may not contain `;` or `,`
* probably shouldn't contain `?:` either, because the `:` would be interpreted as separating dependencies

I think “must only contain plausible characters in a preprocessor expression, and must contain a relation operator” would be a reasonable compromise. How about this (untested)?

```
CONDITION_REGEX = re.compile(
    r'(?!.*\b0[0-9])' # forbid 0ddd which might be accidentally octal or accidentally decimal
    r'(?=.*[!=]=|[<>]=)' # must contain a relation operator
    r'[\s\w!%&()*+-/<=>^|~]+' # plausible characters in an integer constant expression
, re.ASCII)
```

Reply:
Actually I just changed `CONDITION_OPERATOR_REGEX` in `generate_test_code.py` so that now the 2nd element of the comparison can be a 'composition" of symbols, math operators and values. Unfortunately the change still falls into the `framework` subrepo, so I just added one more commit [there](https://github.com/Mbed-TLS/mbedtls-framework/pull/48/commits/fe07e3c9e0dc77191fd67aba57e88f1fd7ecf8e6).